### DISCUSSION TITLE
[Closed] Slow event triggering

### DISCUSSION THREAD

**Discussion by skyeredwood** [Rapptz/discord.py#7553] [Q&A] [answered]
Trying to listen for member update event - intents are working and other events (e.g. ready, message) are triggering but this event in particular won't work. Often it just doesn't work but when it does it's delayed by about 10-20 seconds.

```python
@bot.event
    async def on_member_update(before, after):
        print("Member updated")
```

Any ideas why this is?

**Comment by AmeyWale** [comment, marked as answer]
In order to use this event, the member needs to be cached

**Reply by skyeredwood** [reply]
Ah, thanks! Not very experienced with Python (coming from a NodeJS background), how would I go about that?

**Reply by AmeyWale** [reply]
Make sure that you have turned on guild member intent from developer portal

Include this in your constructor for example
```py
intents = discord.Intents.default()
intents.members = True
client = commands.Bot(command_prefix="your prefix here",intents=intents)

**Reply by skyeredwood** [reply]
Intents already meet that and (just as a test, I understand the risks) am using a selfbot/userbot so can't enable the intent in the dev portal.

**Reply by imvaskel** [reply]
Userbots are not supported by the library anymore, sorry.

**Reply by skyeredwood** [reply]
I'm aware they're not officially supported but surely it shouldn't be an issue anyway - I'm gonna test it with a bot token and see if the issues persist. Hoping they don't however.

**Reply by skyeredwood** [reply]
Update: issues didn't occur when using token for a bot. Real shame as the plan I had in mind won't work without a userbot. Thanks for being helpful anyway.

**Comment by imvaskel** [comment]
Make sure you have the proper intents for the update you are trying to receive (presences / members).
